### PR TITLE
specify parameters to be boxed for recursive types

### LINKED ISSUE #73: recursive types
Services like network have recursive types. See https://github.com/ctaggart/azure-sdk-for-rust/pull/1 for errors and manual workaround. I think we will need to allow outside configuration to box certain fields.

### PR DESCRIPTION
**Pull request by ctaggart**
fix #73
By allowing specific parameters to be boxed for recursive types.

### CODE REVIEW

File: codegen/examples/gen_mgmt.rs [hunk 1]
```diff
@@ -3,61 +3,69 @@
...
+];
+
+// becuse of recursive types, some properties have to be boxed
+// https://github.com/ctaggart/autorust/issues/73
+const BOX_PROPERTIES: &[(&str, &str, &str)] = &[
+    // cost-management
+    ("../azure-rest-api-specs/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2020-06-01/costmanagement.json", "ReportConfigFilter", "not"),
+    ("../azure-rest-api-specs/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2020-06-01/costmanagement.json", "QueryFilter", "not"),
+    // network
+    ("../azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpAddress.json", "PublicIPAddressPropertiesFormat", "ipConfiguration"),
```
Comment (ctaggart) [inline codegen/examples/gen_mgmt.rs:58]: @bmc-msft, if you need to fix a different API version, you can make a pull request to add the entry here.

**Comment by rylev** [comment]
@ctaggart So I guess there's no way to automatically detect which types need to be boxed?

**Comment by ctaggart** [comment]
@rylev, correct.

**Comment by rylev** [comment]
@ctaggart I see a few where we box a type because it has a field which optionally contains itself. We should be able to detect this case no? We just need to look to see if the field's type is the same as the struct.

**Comment by ctaggart** [comment]
@rylev, that case, yes. We can add an GitHub issue for detecting and boxing that automatically. Let's not do it as part of this PR.

**Pull request merged**